File: backend-app/src/Enum/RoleType.php
<?php

namespace App\Enum;

enum RoleType
{
    case APPLICANT;
    case EMPLOYER;
    case INTERIM_AGENCY;
    case MANAGER;

    public static function fromCode(string $code): RoleType
    {
        return match(true) {
            $code === APPLICANT_ROLE_CODE => RoleType::APPLICANT,
            $code === INTERIM_AGENCY_ROLE_CODE => RoleType::INTERIM_AGENCY,
            $code === EMPLOYER_ROLE_CODE => RoleType::EMPLOYER,
            $code === MANAGER_ROLE_CODE => RoleType::MANAGER
        };
    }
}

File: front-end-app/app/src/main/java/com/dev/hirelink/enums/RoleType.kt
package com.dev.hirelink.enums

enum class RoleType(val code: String, val roleName: String) {
    APPLICANT("APP", "APPLICANT"),
    EMPLOYER("EMP", "EMPLOYER"),
    INTERIM_AGENCY("AGC", "AGENCY"),
    MANAGER("MAN", "MANAGER")
}